src/farcasterd/runtime.rs

```diff
@@ -889,6 +895,33 @@ impl Runtime {
                 )?;
             }
 
+            Request::RestoreCheckpoint(swap_id) => {
+                if self.running_swaps.contains(&swap_id) {
+                    endpoints.send_to(
+                        ServiceBus::Ctl,
+                        ServiceId::Farcasterd,
+                        source,
+                        Request::Failure(Failure {
+                            code: 1,
+                            info: "Cannot restore an already running swap".to_string(),
+                        }),
+                    )?;
+                    return Ok(());
```

Comment:
this implies farcasterd Runtime will always have an up to date self.running_swaps

Reply:
Not sure. Maybe send a msg to swapd and if it doesn't error, then swapd is up and running, i guess?

Reply:
It's pringing another ESB error if it send to a non-existing / dead endpoint. I'd like to avoid that.

Reply:
is that true even if we handle the error at the level above internet2? meaning not letting error propagate to internet2

anyway possibly that is something that should end up in internet2 for checking if service is alive

Reply:
Yeah, the error is logged at the ESB level first, before we can capture it in the runtime.

Reply:
I removed the `running_swaps` check and am instead now checking if walletd is still running by sending a `Hello` request on the Msg bus.

Reply:
Good point @zkao. @TheCharlatan & I had a long discussion exploring multiple options for addressing this.
Ultimately, we should have a maintenance daemon that perform aliveness checks with all services. For now, we should just perform this check ad-hoc when restoring a checkpoint - regardless of *yet another ESB error*.

Reply:
Checking if walletd is up in https://github.com/farcaster-project/farcaster-node/pull/477/commits/9734b0f93a0af4a3360a4598544fb43564d55cfe